src/app/api/user/profile/route.ts

```diff
@@ -0,0 +1,113 @@
+import { isString } from "@thi.ng/checks";
+import { NextResponse } from "next/server";
+import { put } from "@vercel/blob";
+import { getUserFromToken, privy } from "@/app/api/utils/auth";
+
+const MAX_IMAGE_SIZE = 2 * 1024 * 1024; // 2MB
+
+/**
+ * Validates the username.
+ * @param {string} username
+ */
+function validateUsername(username) {
```

Comment:
better to have sensitive word detection

Reply:
Makes sense, will discuss this issue with the product. Currently, the name is only visible to the user, and we will optimize it in the future.